google-codelab-survey/google_codelab_survey.js

```diff
@@ -33,33 +68,144 @@ class CodelabSurvey extends HTMLElement {
      * The name of the survey
      * @private {string}
      */
-    this.surveyName_ = this.getAttribute('survey-id') || '';
+    this.surveyName_ = this.getAttribute(SURVEY_ID_ATTR) || DEFAULT_SURVEY_NAME;
+
+    /** @private {!HTML5LocalStorage} */
+    this.storage_ = new HTML5LocalStorage();
+
+    /** @private {string} */
+    this.storageKey_ = STORAGE_KEY_PREFIX + this.surveyName_;
+
+    /** @private {!Object<string, !Object>} */
+    this.storedData_ = {};
+
+    /** @private {!EventHandler} */
+    this.eventHandler_ = new EventHandler();
   }
 
   /**
    * @export
    * @override
    */
   connectedCallback() {
+    this.checkStoredData_();
     this.updateDom_();
+    this.bindEvents_();
+  }
+
+  /** @private */
+  bindEvents_() {
+    this.eventHandler_.listen(document.body, events.EventType.CLICK, (e) => {
```

Comment:
You can also do:
this.eventHandler_.listen(document.body, events.EventType.CLICK,
    (e) => this.handleClick_(e.target));

Reply:
Done.